### DISCUSSION TITLE
How can I debug my own Chrome extension in the Theia framework? I noticed that background.js doesn't seem to be working.

### DISCUSSION THREAD

**Discussion by fjxfirst** [eclipse-theia/theia#14049] [Q&A]
I developed a Chrome extension that is a DevTools extension, similar to the Network panel, used to record and manage frontend-to-backend data interactions to improve team development efficiency. However, it doesn't handle HTTP requests; instead, it works with RPC. The extension injects scripts using content.js, and methods are called within the project to send messages to background.js. However, background.js doesn't seem to be working. I can confirm that my Chrome extension loads and works correctly in a real browser.

In my Theia project, when I open the Developer Tools, I can see that my Chrome extension has successfully loaded, and the panel interface is visible. However, when I right-click on the panel page, the context menu does not appear as it does in a real Chrome browser, so I can't inspect the page. Additionally, I can't find an entry point to inspect the Service Worker or the background.js, making it impossible to debug the background.js code.

Environment Information:

Theia Version: 1.48.0
Node Version: 18.18.0
```
import { session } from '@theia/core/electron-shared/electron';
session.defaultSession.loadExtension('c:\\xxx\\xxx\\devtool-demo', { allowFileAccess: true });
```

manifest.json
```json
{
  "manifest_version": 3,
  "name": "request-demo",
  "version": "0.0.1",
  "description": "xxxx",
  "background": {
    "service_worker": "js/background.js",
    "type": "module"
  },
  "devtools_page": "devtools.html",
  "permissions": ["storage", "scripting", "tabs"],
  "icons": {
    "16": "icon.png",
    "32": "icon.png",
    "48": "icon.png",
    "128": "icon.png"
  },
  "content_scripts": [
    {
      "matches": ["<all_urls>"],
      "js": ["js/content.js"],
      "run_at": "document_idle"
    }
  ],
  "web_accessible_resources": [
    {
      "resources": ["js/inject.js"],
      "matches": ["*://*/*"]
    }
  ],
  "host_permissions": ["<all_urls>"]
}
```

**Comment by msujew** [comment]
Hey @fjxfirst, just to explain why no one is answering - you're not being ignored, I just don't think anyone from the Theia team has any kind of experience working with Chrome extensions, me included.
